village/src/components/SmurfForm.js

```diff
@@ -1,34 +1,61 @@
 import React, { Component } from 'react';
+import { NavLink } from 'react-router-dom';
 
 class SmurfForm extends Component {
   constructor(props) {
     super(props);
     this.state = {
       name: '',
       age: '',
-      height: ''
+      height: '',
+      id: ''
     };
   }
 
+  componentDidMount = () => {
+    const {id, name} = this.state;
+    console.log(id);
+    console.log(name);
+  }
+
   addSmurf = event => {
     event.preventDefault();
     // add code to create the smurf using the api
+    const { name, age, height } = this.state;
+    this.props.addSmurf({ name: name, age: age, height: height });
 
     this.setState({
       name: '',
       age: '',
-      height: ''
+      height: '',
+      id: ''
+    });
+  }
+
+  updateSmurf = event => {
+    event.preventDefault();
+    const { id, name, age, height } = this.state;
+    this.props.updateSmurf( id, { name: name, age: age, height: height });
+
+    this.setState({
```

Comment:
Instead of typing out the whole object you could just send the whole state. So your function will look something like 
```
this.props.updateSmurf( id, this.state);
```